src/firmware/app/primitives/move_helper.c

```diff
@@ -18,125 +18,32 @@
 // so that the axes would never have to compete for resources
 #define TIME_HORIZON 0.05f  // s
 
-// The minimum distance away from our destination that we must be if we
-// are going to rotate the bot onto its wheel axis
-// 2 * P_PI * ROBOT_RADIUS = robot circumference, which is approximately
-// how far the bot would have to turn for one full rotation, so we
-// set it a litle larger than that.
-const float APPROACH_LIMIT = 3 * P_PI * ROBOT_RADIUS;
-
-const float PI_2 = P_PI / 2.0f;
-
 typedef struct MoveHelperState
 {
-    float destination[3], end_speed, major_vec[2], minor_vec[2];
-    // We store a wheel index here so we only have to calculate the axis
-    // we want to use when move start is called
-    unsigned optimal_wheel_axes_index;
-    bool slow;
-} MoveHelperState_t;
-DEFINE_PRIMITIVE_STATE_CREATE_AND_DESTROY_FUNCTIONS(MoveHelperState_t)
+    // The trajectory we're tracking
+    PositionTrajectory_t position_trajectory;
 
-/**
- * Call from move_start to choose which wheel axis we will be
- * using for preliminary rotation. The idea is to pick the wheel
- * axis that will result in the minimum remaining rotation onto
- * the bot's final destination angle.
- *
- * @param dx the global x position of the bot
- * @param dy the global y position of the bot
- * @param current_angle the current angle of the bot
- * @param final_angle the final destination angle
- * @return the index of the wheel axis to use
- */
-unsigned choose_wheel_axis(float dx, float dy, float current_angle, float final_angle);
+    // The number of elements in the trajectory we're tracking
+    size_t num_trajectory_elems;
+
+    // The start time of this primitive, in seconds
+    float primitive_start_time_seconds;
+
+    // Whether or not we're trying to move slowly
+    bool move_slow;
+} MoveHelperState_t;
```

Comment:
should rename movehelper to something related to trajectory planner

Reply:
Why do you say that? As I understand it this represents the state for the "Move Helper"?

Reply:
I meant that **`MoveHelper`** (not MoveHelperState_t) isn't a good name given that it uses the trajectory planner, so we can be more specific here

Reply:
I agree with both y'all. I don't know what we would really name it though. It's just a further abstraction around the TP data